elide-core/src/main/java/com/yahoo/elide/core/request/route/PathRouteResolver.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2023, the original author or authors.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core.request.route;
+
+import static com.yahoo.elide.core.dictionary.EntityDictionary.NO_VERSION;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Resolves a route using the path.
+ */
+public class PathRouteResolver implements RouteResolver {
+
+    private final String versionPrefix;
+    private final ApiVersionValidator apiVersionValidator;
+
+    public PathRouteResolver(String versionPrefix, ApiVersionValidator apiVersionValidator) {
+        this.versionPrefix = versionPrefix;
+        this.apiVersionValidator = apiVersionValidator;
+    }
+
+    @Override
+    public Route resolve(String mediaType, String baseUrl, String path,
+            Map<String, List<String>> headers, Map<String, List<String>> parameters) {
+        String baseRoute = baseUrl == null ? "" : baseUrl;
+        String route = path;
+        String apiVersion = NO_VERSION;
+
+        String apiVersionString = "";
+        int versionStart = -1;
+        int versionEnd = -1;
+        int pathStart = -1;
+        int pathEnd = -1;
+
+        int find = path.indexOf('/', 0);
+        if (find != -1) {
```

Comment:
I think this could be simpler with a regex.

Reply:
While I didn't profile it, since this essentially called for every request, I thought that doing string operations might be faster than a regex.

Reply:
Ok.  The logic looks correct to me.